src/bridge/events.rs

```diff
@@ -493,9 +493,25 @@ fn parse_style(style_map: Value) -> Result<Style> {
                 ("strikethrough", Value::Boolean(strikethrough)) => {
                     style.strikethrough = strikethrough
                 }
-                ("underline", Value::Boolean(underline)) => style.underline = underline,
-                ("undercurl", Value::Boolean(undercurl)) => style.undercurl = undercurl,
                 ("blend", Value::Integer(blend)) => style.blend = blend.as_u64().unwrap() as u8,
+
+                // TODO: handle correct precedence?
+                ("underline", Value::Boolean(true)) => {
+                    style.underline = Some(UnderlineStyle::Underline)
+                }
+                ("undercurl", Value::Boolean(true)) => {
+                    style.underline = Some(UnderlineStyle::UnderCurl)
+                }
+                ("underdotted", Value::Boolean(true)) => {
+                    style.underline = Some(UnderlineStyle::UnderDot)
+                }
+                ("underdashed", Value::Boolean(true)) => {
+                    style.underline = Some(UnderlineStyle::UnderDash)
+                }
+                ("underdouble", Value::Boolean(true)) => {
```

Comment:
`:h attr-list` calls those `underdot`, `underdash` and `underlineline`, do you have any place where the `ed` suffix comes from? I also like `double` more than `lineline`, but I don't want to break compatibility with the rest of the ecosystem due to poor naming.

```suggestion
                ("underdot", Value::Boolean(true)) => {
                    style.underline = Some(UnderlineStyle::UnderDot)
                }
                ("underdash", Value::Boolean(true)) => {
                    style.underline = Some(UnderlineStyle::UnderDash)
                }
                ("underlineline", Value::Boolean(true)) => {
```

Reply:
As mentioned in my other comment, the latest neovim versions seem to have changed the names so that they become aligned with vim's names. I will amend this to support both names so that people aren't forced to upgrade.